searches/search_test.go

```diff
@@ -1,49 +1,89 @@
 package searches
 
-import "testing"
+import (
+	"testing"
+)
 
 type searchTest struct {
-	data     []int
-	key      int
-	expected int
-	name     string
+	data          []int
+	key           int
+	expectedValue int
+	expectedError error
+	name          string
 }
 
 var searchTests = []searchTest{
 	//Sanity
-	{[]int{1, 2, 3, 4, 5, 6, 7, 8, 9, 10}, 1, 0, "Sanity"},
-	{[]int{1, 2, 3, 4, 5, 6, 7, 8, 9, 10}, 5, 4, "Sanity"},
-	{[]int{1, 2, 3, 4, 5, 6, 7, 8, 9, 10}, 10, 9, "Sanity"},
+	{[]int{1, 2, 3, 4, 5, 6, 7, 8, 9, 10}, 10, 9, nil, "Sanity"},
+	{[]int{1, 2, 3, 4, 5, 6, 7, 8, 9, 10}, 9, 8, nil, "Sanity"},
+	{[]int{1, 2, 3, 4, 5, 6, 7, 8, 9, 10}, 8, 7, nil, "Sanity"},
+	{[]int{1, 2, 3, 4, 5, 6, 7, 8, 9, 10}, 7, 6, nil, "Sanity"},
+	{[]int{1, 2, 3, 4, 5, 6, 7, 8, 9, 10}, 6, 5, nil, "Sanity"},
+	{[]int{1, 2, 3, 4, 5, 6, 7, 8, 9, 10}, 5, 4, nil, "Sanity"},
+	{[]int{1, 2, 3, 4, 5, 6, 7, 8, 9, 10}, 4, 3, nil, "Sanity"},
+	{[]int{1, 2, 3, 4, 5, 6, 7, 8, 9, 10}, 3, 2, nil, "Sanity"},
+	{[]int{1, 2, 3, 4, 5, 6, 7, 8, 9, 10}, 2, 1, nil, "Sanity"},
+	{[]int{1, 2, 3, 4, 5, 6, 7, 8, 9, 10}, 1, 0, nil, "Sanity"},
 	//Absent
-	{[]int{1, 4, 5, 6, 7, 10}, -25, -1, "Absent"},
-	{[]int{1, 4, 5, 6, 7, 10}, 25, -1, "Absent"},
+	{[]int{1, 4, 5, 6, 7, 10}, -25, -1, errNotFound, "Absent"},
+	{[]int{1, 4, 5, 6, 7, 10}, 25, -1, errNotFound, "Absent"},
 	//Empty slice
-	{[]int{}, 2, -1, "Empty"},
+	{[]int{}, 2, -1, errNotFound, "Empty"},
```

Comment:
Could you move the test cases inside the test to make sure that there is no conflict with new searching algorithms that are created in this package. Also, do a git pull from our master branch and resolve conflicts. We have changed the structure of the repository which would create conflicts for this PR. For example, you can see that the package name has been changed to `search` instead of `searches` and same applies for the directory. Let me know when you are ready for this PR to be reviewed. If you do not interact with this PR for two weeks then I would consider this PR to be stale and close it. You will be able to reopen it in the case that you did not see this message.

Reply:
The new structure have a single `testcases.go` file share across tests for different search functions. Since currently `Binary` and `BinaryIterative` function return error for when value not found but the other search functions did not, I have opted to ignore error on the test for now. Should I add a error return value for other searches?

Reply:
@tjgurwara99